lib/Sema/DerivedConformanceAdditiveArithmeticVectorNumeric.cpp

```diff
@@ -98,6 +98,8 @@ static ConstructorDecl *getMemberwiseInitializer(NominalTypeDecl *nominal) {
 static Type getVectorNumericScalarAssocType(ValueDecl *decl) {
   auto &C = decl->getASTContext();
   auto *vectorNumericProto = C.getProtocol(KnownProtocolKind::VectorNumeric);
+  if (!decl->hasInterfaceType())
```

Comment:
Could you use one of the following two snippets instead?

1. From [`DerivedConformanceCodable.cpp`](https://github.com/apple/swift/blob/c360c82850c884b32dad617ec91028803e5ed4f2/lib/Sema/DerivedConformanceCodable.cpp#L133):
```cpp
if (!varDecl->hasType())
  tc.validateDecl(varDecl);
if (!varDecl->hasType())
  return false;
```

Use this when you have access to a `TypeChecker`.

2. From [`DerivedConformanceEquatableHashable.cpp`](https://github.com/apple/swift/blob/eb75ad80dcfdc21e7ee1954d2c66c2c11fdee04d/lib/Sema/DerivedConformanceEquatableHashable.cpp#L76):
```cpp
// auto *lazyResolver = C.getLazyResolver();
if (!propertyDecl->hasType())
  lazyResolver->resolveDeclSignature(propertyDecl);
if (!propertyDecl->hasType())
  return false;
```

Use this when you don't have access to a `TypeChecker`, only an `ASTContext`.

---

So here, please use the first snippet.

Reply:
Done